gcc error when building matplotlib dev from source

<!--To help us understand and resolve your issue, please fill out the form to the best of your ability.-->
<!--You can feel free to delete the sections that do not apply.-->

### Bug report

**Bug summary**

Failed to build dev version of `matplotlib` from source. Example log: https://travis-ci.org/astropy/astropy/jobs/538721543

```
set build/lib.linux-x86_64-3.7/matplotlib/_version.py to '3.1.0+955.ge2b581abb'
    running build_ext
    building 'matplotlib.ft2font' extension
    creating build/temp.linux-x86_64-3.7
    creating build/temp.linux-x86_64-3.7/src
    gcc -DNDEBUG -g -fwrapv -O3 -Wall -std=c99 -fPIC -DFREETYPE_BUILD_TYPE=system -DPY_ARRAY_UNIQUE_SYMBOL=MPL_matplotlib_ft2font_ARRAY_API -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -D__STDC_FORMAT_MACROS=1 -Iextern/agg24-svn/include -I/home/travis/miniconda/envs/test/lib/python3.7/site-packages/numpy/core/include -I/home/travis/miniconda/envs/test/include/python3.7m -c src/checkdep_freetype2.c -o build/temp.linux-x86_64-3.7/src/checkdep_freetype2.o -I/usr/include/freetype2
    src/checkdep_freetype2.c:1:45: error: missing binary operator before token "("
     #if defined __has_include && ! __has_include(<ft2build.h>)
                                                 ^
    src/checkdep_freetype2.c:15:11: note: #pragma message: Compiling with FreeType version 2.5.2.
       #pragma message("Compiling with FreeType version " \
               ^
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command "/home/travis/miniconda/envs/test/bin/python -u -c 'import setuptools, tokenize;__file__='"'"'/tmp/pip-install-cbwrywg3/matplotlib/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-hmjgqamm/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-install-cbwrywg3/matplotlib/
```

We did not see this error when CI ran 20 hours ago; https://travis-ci.org/astropy/astropy/jobs/538455547 failed with something else that was unrelated to `matplotlib`. But definitely was happening 4 hours ago (https://travis-ci.org/astropy/astropy/jobs/538718110).

<!--A short 1-2 sentences that succinctly describes the bug-->

**Code for reproduction**

<!--A minimum code snippet required to reproduce the bug.
Please make sure to minimize the number of dependencies required, and provide
any necessary plotted data.
Avoid using threads, as Matplotlib is (explicitly) not thread-safe.-->

Here is how our CI pick up dev version of `matplotlib`:

https://github.com/astropy/ci-helpers/blob/d9cd38228f30cf939bb47ef5e07175421f87ac33/travis/setup_dependencies_common.sh#L605-L617

**Actual outcome**

<!--The output produced by the above code, which may be a screenshot, console output, etc.-->

See logs posted in summary above.

**Expected outcome**

<!--A description of the expected outcome from the code snippet-->
<!--If this used to work in an earlier version of Matplotlib, please note the version it used to work on-->

Successful build of `matplotlib` dev.

**Matplotlib version**
<!--Please specify your platform and versions of the relevant libraries you are using:-->
  * Operating system: Ubuntu 4.8.4-2ubuntu1~14.04.3
  * Matplotlib version: 3.1.0+955.ge2b581abb
  * Matplotlib backend (`print(matplotlib.get_backend())`): Agg (https://github.com/astropy/astropy/blob/62f1ac1726e57facf922eab9fa3ee36b03aa8a2d/astropy/conftest.py#L37)
  * Python version: 3.6 and 3.7
  * Jupyter version (if applicable): N/A
  * Other libraries: https://github.com/astropy/astropy/blob/62f1ac1726e57facf922eab9fa3ee36b03aa8a2d/.travis.yml#L38

<!--Please tell us how you installed matplotlib and python e.g., from source, pip, conda-->
<!--If you installed from conda, please specify which channel you used if not the default-->

--- a/src/checkdep_freetype2.c
+++ b/src/checkdep_freetype2.c
@@ -1,23 +1,21 @@
-#if defined __has_include && ! __has_include(<ft2build.h>)
-
-  #error "FreeType version 2.3 or higher is required. \
-You may set the MPLLOCALFREETYPE environment variable to 1 to let \
-Matplotlib download it."
-
-#else
-
-  #include <ft2build.h>
-  #include FT_FREETYPE_H
-
-  #define XSTR(x) STR(x)
-  #define STR(x) #x
-
-  #pragma message("Compiling with FreeType version " \
-    XSTR(FREETYPE_MAJOR) "." XSTR(FREETYPE_MINOR) "." XSTR(FREETYPE_PATCH) ".")
-  #if FREETYPE_MAJOR << 16 + FREETYPE_MINOR << 8 + FREETYPE_PATCH < 0x020300
+#ifdef __has_include
+  #if !__has_include(<ft2build.h>)
     #error "FreeType version 2.3 or higher is required. \
-You may set the MPLLOCALFREETYPE environment variable to 1 to let \
-Matplotlib download it."
+You may set the MPLLOCALFREETYPE environment variable to 1 to let Matplotlib \
+download it."
   #endif
+#endif
 
+#include <ft2build.h>
+#include FT_FREETYPE_H
+
+#define XSTR(x) STR(x)
+#define STR(x) #x
+
+#pragma message("Compiling with FreeType version " \
+  XSTR(FREETYPE_MAJOR) "." XSTR(FREETYPE_MINOR) "." XSTR(FREETYPE_PATCH) ".")
+#if FREETYPE_MAJOR << 16 + FREETYPE_MINOR << 8 + FREETYPE_PATCH < 0x020300
+  #error "FreeType version 2.3 or higher is required. \
+You may set the MPLLOCALFREETYPE environment variable to 1 to let Matplotlib \
+download it."
 #endif

--- a/src/checkdep_libpng.c
+++ b/src/checkdep_libpng.c
@@ -1,19 +1,17 @@
-#if defined __has_include && ! __has_include(<png.h>)
-
-  #error "libpng version 1.2 or higher is required. \
-Consider installing it with e.g. 'conda install libpng', \
-'apt install libpng12-dev', 'dnf install libpng-devel', or \
-'brew install libpng'."
-
-#else
-
-  #include <png.h>
-  #pragma message("Compiling with libpng version " PNG_LIBPNG_VER_STRING ".")
-  #if PNG_LIBPNG_VER < 10200
+#ifdef __has_include
+  #if !__has_include(<png.h>)
     #error "libpng version 1.2 or higher is required. \
 Consider installing it with e.g. 'conda install libpng', \
 'apt install libpng12-dev', 'dnf install libpng-devel', or \
 'brew install libpng'."
   #endif
+#endif
 
+#include <png.h>
+#pragma message("Compiling with libpng version " PNG_LIBPNG_VER_STRING ".")
+#if PNG_LIBPNG_VER < 10200
+  #error "libpng version 1.2 or higher is required. \
+Consider installing it with e.g. 'conda install libpng', \
+'apt install libpng12-dev', 'dnf install libpng-devel', or \
+'brew install libpng'."
 #endif

